### PR TITLE
[dart-dio] Fixes errors when primitive return types are attempted to …

### PR DESCRIPTION
**Pull request by josh-burton**
…be deserialized as json

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

CC: @ircecho @swipesight @jaumard @amondnet

### CODE REVIEW

**Comment by kuhnroyal** [comment]
@josh-burton This needs to be `{{{returnType}}}` or it will break with primitive lists.

**Comment by josh-burton** [comment]
@kuhnroyal whats the difference between `{{returnType}}` and `{{{returnType}}}`? (has always confused me)

**Comment by kuhnroyal** [comment]
Afaik 2 braces perform HTML escaping and won't work for types like `List<String>` that's why you mostly see 3 braces for type definitions.

Can you rebase the branch and see if it goes green? So many small fixes for Dart but they don't get merged.

**Comment by kuhnroyal** [comment]
@josh-burton What is missing to get this merged?

**Comment by josh-burton** [comment]
@kuhnroyal I'm still waiting on this to be reviewed. CC @ircecho @swipesight @jaumard @amondnet

**Comment by kuhnroyal** [comment]
@josh-burton I fixed the same problem in https://github.com/OpenAPITools/openapi-generator/pull/8025 but your fix makes more sense so I will rebase once this here is merged.

**Comment by kuhnroyal** [comment]
My branch with integration test got merged first.
You can now run integration tests with `./mvnw integration-test -f samples/client/petstore/dart-dio/pom.xml`

**Comment by kuhnroyal** [comment]
@josh-burton Can you rebase this please? We are currently merging a lot of PRs so I am sure we can get this in.

File: samples/openapi3/client/petstore/dart-dio/petstore_client_lib_fake/lib/model/enum_class.dart [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import 'package:built_collection/built_collection.dart';
+import 'package:built_value/built_value.dart';
+import 'package:built_value/serializer.dart';
+
+part 'enum_class.g.dart';
+
+class EnumClass extends EnumClass {
```
Comment (kuhnroyal) [inline samples/openapi3/client/petstore/dart-dio/petstore_client_lib_fake/lib/model/enum_class.dart:7]: There is something wrong here when these types get generated - this was fixed in 7fae4ab4116c826ca64921babd239fbf1eb2d180

Comment (josh-burton) [reply samples/openapi3/client/petstore/dart-dio/petstore_client_lib_fake/lib/model/enum_class.dart:7]: yep just fixing that :)

Comment (kuhnroyal) [reply samples/openapi3/client/petstore/dart-dio/petstore_client_lib_fake/lib/model/enum_class.dart:7]: You need to delete the 4 old files manually.

**Review by kuhnroyal — APPROVED** [review]

**Comment by wing328** [comment]
CircleCI failure not related to this change.

**Pull request merged**